charts/gitness/templates/deployment.yaml

```diff
@@ -0,0 +1,95 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "gitness.fullname" . }}
+  labels:
+    {{- include "gitness.labels" . | nindent 4 }}
+spec:
+  {{- if not .Values.autoscaling.enabled }}
+  replicas: {{ .Values.replicaCount }}
```

Comment:
currently multiple pods of gitness are not possible. We should hardcode this to 1.